pkg/planner/core/indexmerge_unfinished_path.go

```diff
@@ -135,24 +135,15 @@ func initUnfinishedPathsFromExpr(
 		ret[i].index = path.Index
 		// case 1: try to use the previous logic to handle non-mv index
 		if !isMVIndexPath(path) {
-			// generateNormalIndexPartialPaths4DNF is introduced for handle a slice of DNF items and a slice of
-			// candidate AccessPaths before, now we reuse it to handle single filter and single candidate AccessPath,
-			// so we need to wrap them in a slice here.
 			partialPath, needSelection := generateNormalIndexPartialPath(
 				ds,
 				expr,
 				path,
 			)
 			if partialPath != nil {
 				ret[i].initedWithValidRange = true
-				ret[i].usableFilters = partialPath.AccessConds
 				ret[i].needKeepFilter = needSelection
-				// Here is a special case, if this expr is always false and this path is a dual path, it will run to
-				// this point, and paths[0].AccessConds and paths[0].Ranges will be nil.
-				// In this case, we set the accessFilters to the original expr.
-				if len(ret[i].usableFilters) <= 0 {
-					ret[i].usableFilters = []expression.Expression{expr}
-				}
+				ret[i].usableFilters = []expression.Expression{expr}
```

Comment:
The changes here should have been included in the last PR but I missed it.
It turns out we don't need, and also should not differentiate the two cases. Otherwise, the final plan may miss some IndexFilters.